msal/src/Microsoft.Identity.Client/TokenCache.cs

```diff
@@ -325,9 +325,21 @@ internal async Task<MsalAccessTokenCacheItem> FindAccessTokenAsync(Authenticatio
         private MsalAccessTokenCacheItem FindAccessTokenCommon
             (AuthenticationRequestParameters requestParams, string preferredEnvironmentAlias, ISet<string> environmentAliases)
         {
+            string msg;
+
+            //no authority passed
+            if (environmentAliases.Count == 0)
+            {
+                msg = "No authority provided. Skipping cache lookup ";
```

Comment:
>msg [](start = 16, length = 3)

While issue exists everywhere, pls. consider making such strings consts.

We are passing strings around which means it will continue to be copied into memory causing extra CPU cycles

Reply:
it is just local log message it is is not reused

Reply:
It may or may only be used once, but right now you are passing it to 2 different methods both constructing a copy of the string. had you however used a stringbuilder it would have been by reference. alternatively you can use a const, if I recall correctly it will not create the extra copy (which uses memory and CPU cycles)

---
In reply to: [216103745](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1246#discussion_r216103745) [](ancestors = 216103745)